### PR TITLE
Publish to hex.pm

### LINKED ISSUE #223: Publish to hex.pm [JIRA: RIAK-3280]
See basho/riak-erlang-client#238 and basho/riak-erlang-client#352

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #223 (RIAK-1897) (RIAK-3280) 

See basho/riak-erlang-client#238 (RIAK-1401) (RIAK-1892) (RIAK-3222) and basho/riak-erlang-client#352 (RIAK-2543)

Edited this page as well: https://github.com/basho/riak_pb/wiki

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,11 +42,20 @@ endif
...
+endif
+	echo "==> Tagging version $(VERSION)"
+	$(PROJDIR)/build/publish $(VERSION) validate
+	echo -n "$(VERSION)" > VERSION
+	git add --force VERSION
+	git commit --message="riak_pb $(VERSION)"
+	git push
+	git tag --sign -a "$(VERSION)" -m "riak_pb $(VERSION)" --local-user "$(RELEASE_GPG_KEYNAME)"
+	git push --tags
+	$(PROJDIR)/build/publish $(VERSION)
```
Comment (javajolt) [inline Makefile]: Looks like at least there are some comments in the `publish` script which mention the requirement of an `X.Y.Z.N` version string

File: VERSION [hunk 2]
```diff
@@ -0,0 +1 @@
+2.3.0-1-beta1
```
Comment (alexmoore) [inline VERSION]: Just for mechanism testing purposes.

File: build/publish [hunk 3]
```diff
@@ -115,13 +115,13 @@ trap onexit EXIT
 
 declare -r version_string="${1:-unknown}"
 
-if [[ ! $version_string =~ ^[0-9].[0-9].[0-9](.[0-9])?(-[a-z]+[0-9]+)?$ ]]
+if [[ ! $version_string =~ ^[0-9].[0-9].[0-9](-[0-9]+)?(-[a-z]+[0-9]+)?$ ]]
 then
     errexit 'first argument must be valid version string in X.Y.Z.N format'
```
Comment (alexmoore) [inline build/publish]: This'll have to change to "X.Y.Z-N format".

File: mix.exs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  defp deps do
+    [
+      {:hamcrest, "~> 0.4.0-beta2-basho", hex: :basho_hamcrest},
+      {:ex_doc, ">= 0.0.0", only: :dev}
+    ]
+  end
+
+  defp package do
+    [files: ~w(mix.exs src plugins include rebar.config tools.mk README.md VERSION LICENSE Makefile),
+     maintainers: ["Drew Kerrigan, Luke Bakken"],
```
Comment (alexmoore) [inline mix.exs]: Indeed 😄

File: mix.exs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    catch
+      _ ->
+        []
+    end
+    [:debug_info, :warnings_as_errors | extra_options]
+  end
+
+  defp deps do
+    [
+      {:hamcrest, "~> 0.4.0-beta2-basho", hex: :basho_hamcrest},
```
Comment (alexmoore) [inline mix.exs]: Yeah, when we cut the actual version.

File: VERSION (hunk 2 above)
Comment (lukebakken) [inline VERSION]: FYI, `VERSION` is updated by the `make release` command.

File: Makefile (hunk 1 above)
Comment (lukebakken) [inline Makefile]: Thanks for noticing, got it

**Pull request merged**